pkg/storage/common.go

```diff
@@ -91,6 +92,16 @@ func ValidateManifest(imgStore ImageStore, repo, reference, mediaType string, bo
 				return "", zerr.ErrBadManifest
 			}
 		}
+
+		// a lightweight check if the blob is present
+		for _, layer := range manifest.Layers {
+			blobPath := imgStore.BlobPath(repo, layer.Digest)
+			if _, err := os.Stat(blobPath); err != nil {
```

Comment:
This is common code for both local and s3. This does not work. You need to call an image store API.